src/network/networkd-setlink.c

```diff
@@ -502,6 +502,14 @@ static int link_is_ready_to_set_link(Link *link, Request *req) {
                         r = link_down_now(link);
                         if (r < 0)
                                 return r;
+
+                        /* If the kind of the link is "bond", we need
+                         * set the slave link down as well. */
+                        if (link->kind && streq_ptr(link->kind, "bond")) {
```

Comment:
`streq_ptr()` gracefully handles the case that `link->kind` is `NULL`. Hence, please simply do like the following:
```c
if (streq_ptr(link->kind, "bond")) {
```